Ark.Tools.Outbox.SqlServer/SqlContextAsyncFactory.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Ark.Tools.Core;
+using Ark.Tools.Sql;
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Ark.Tools.Outbox.SqlServer
+{
+    public abstract class SqlContextAsyncFactory<T> : IContextFactory<T> where T : ISqlContextAsync, IAsyncDisposable
+    {
+        private readonly IDataContextConfig _contextConfig;
+        private readonly IDbConnectionManager _dbConnectionManager;
+        private IsolationLevel _isolationLevel;
+        private DbConnection? _dbConnection;
```

Comment:
cmon! I just commented that a factory with private disposables doesn't make sense..